src/i_oalsound.c

```diff
@@ -675,19 +645,18 @@ boolean I_OAL_CacheSound(sfxinfo_t *sfx)
 
             sampledata = lumpdata + DMXHDRSIZE;
 
-            if (IsPaddedSound(sampledata, size))
-            {
-                // Ignore DMX padding.
-                sampledata += DMXPADSIZE;
-                size -= DMXPADSIZE * 2;
-            }
+            // DMX skips the first and last 16 bytes of data. Custom sounds may
+            // be created with tools that aren't aware of this, which means part
+            // of the waveform is cut off. We compensate for this by fading in
+            // or out sounds that start or end at a non-zero amplitude to
+            // prevent clicking.
+            // Reference: https://www.doomworld.com/forum/post/949486
+            sampledata += DMXPADSIZE;
+            size -= DMXPADSIZE * 2;
```

Comment:
Pedantic, but how about zero size lumps?

Reply:
That's guarded already a few lines up.